src/frontend/src/optimizer/rule/distinct_agg.rs

```diff
@@ -183,11 +186,9 @@ impl DistinctAggRule {
                 agg_call.filter = Condition::true_cond();
 
                 // change final agg's agg_kind just like two-phase agg.
-                match agg_call.agg_kind {
-                    AggKind::Count | AggKind::ApproxCountDistinct => {
-                        agg_call.agg_kind = AggKind::Sum;
-                    }
-                    _ => {}
+                if agg_call.agg_kind == AggKind::Count {
```

Comment:
Similarity, future AggKind may or may not be able to use the same agg call for mid and final agg. How about writing an exhaustive match (i.e. without `_`) here so that the compiler would remind the people adding new AggKind to update this place?